org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractPosition.java

```diff
@@ -60,7 +63,8 @@ public int hashCode() {
 		final int prime = 31;
 		int result = 1;
 		long temp;
-		final Collection<String> names   = getNames();
+		final List<String> names = new ArrayList<String>(getNames());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)